source/common/api/posix/os_sys_calls_impl.cc

```diff
@@ -95,6 +95,29 @@ bool OsSysCallsImpl::supportsUdpGro() const {
 #endif
 }
 
+bool OsSysCallsImpl::supportsUdpGso() const {
+#if !defined(__linux__)
+  return false;
+#else
+#ifndef UDP_SEGMENT
+  return false;
+#else
```

Comment:
Detecting whether UDP_SEGMENT is defined at compiling time doesn't do the job correctly. UDP_SEGMENT isn't defined in glibc in Ubuntu 18.04 while its kernel supports it. Same applies for `UDP_GRO` above.
The coverage report clearly shows this is always returning false: https://storage.googleapis.com/envoy-pr/12219/coverage/source/common/api/posix/os_sys_calls_impl.cc.gcov.html

We should define UDP_SEGMENT if it's not defined like: https://quiche.googlesource.com/quiche/+/refs/heads/master/quic/core/quic_linux_socket_utils.h#31

@danzh2010

Reply:
Thanks for pointing this out! I am now defining both `UDP_SEGMENT` and `UDP_GRO` under `include/envoy/common/platform.h`, similar to the way it is done under the quiche source code.